Sources/Plasma/NucleusLib/pnUUID/pnUUID_Unix.cpp

```diff
@@ -50,41 +50,30 @@ static_assert(sizeof(plUUID) == sizeof(uuid_t), "plUUID and uuid_t types differ
 
 struct plUUIDHelper
 {
-    typedef struct _GUID {
-      uint32_t Data1;
-      uint16_t Data2;
-      uint16_t Data3;
-      uint8_t  Data4[8];
-    } GUID;
-    
     static inline void CopyToPlasma( plUUID * dst, const uuid_t & src )
     {
-        hsAssert( sizeof(uuid_t)==sizeof(dst->fData), "sizeof(uuid_t)!=sizeof(plUUID)" );
-        memcpy( (void*)dst->fData, (const void *)src, sizeof( plUUID ) );
-        
-        /*
-         GUIDs are always internally encoded in little endian,
-         while the uuid_t structure is big endian.
-         */
-        GUID* guid = reinterpret_cast<GUID*>(dst->fData);
-        guid->Data1 = hsSwapEndian32(guid->Data1);
-        guid->Data2 = hsSwapEndian16(guid->Data2);
-        guid->Data3 = hsSwapEndian16(guid->Data3);
+        hsAssert(sizeof(uuid_t) == sizeof(dst->fData), "sizeof(uuid_t) != sizeof(plUUID)");
+        memcpy((void*)dst->fData, (const void*)src, sizeof(plUUID));
```

Comment:
The casts shouldn't be necessary here. Even in C++, every pointer can be implicitly converted to `void*` (as long as you obey `const`, etc.).

The `src` cast is especially worrying to me. `src` is a `uuid_t&` *reference*, not a pointer, so I'm not sure how `(const void*)src` could work at all...

```suggestion
        memcpy(dst->fData, &src, sizeof(plUUID));
```

Reply:
`uuid_t` appears to be a typedef for `unsigned char[12]`